examples/rectangle_area.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+
+from pyshgp.gp.estimators import PushEstimator
+from pyshgp.gp.genome import GeneSpawner
+from pyshgp.push.interpreter import PushInterpreter
+from pyshgp.push.instruction_set import InstructionSet
+from pyshgp.push.instruction import SimpleInstruction
+from pyshgp.push.type_library import PushTypeLibrary
+from pyshgp.push.types import PushFloat
+
+"""
+In this demo, we attempt to evolve a program that can find the difference between the areas of two rectangles.
+This is a demonstration of how to create your own PushTypes and PushInstructions.
+"""
+
+"""
+We begin by creating a Rectangle class. 
+Each Rectangle has an float width and height.
+This class will be the underlying type for our custom PushType.
+"""
+
+
+class Rectangle:
+
+    # constructor
+    def __init__(self, height: float, width: float):
+        self.height = height
+        self.width = width
+
+    # checks the equality of two Rectangle objects
+    def __eq__(self, other):
+        if isinstance(other, Rectangle):
+            return self.height == other.height and self.width == other.width
+        return False
+
+    # returns the Rectangle object's representation
```

Comment:
The comments before each method don't tell us much that we don't get from the name of the method. I think the code would be slightly cleaner without them.